scalyr_agent/agent_main.py

```diff
@@ -1069,7 +1069,527 @@ def __get_log_files_initial_positions(self, only_new_files=False):
+        """Runs the Scalyr Agent 2.
+
+        This method will not return until a TERM signal is received or a fatal error occurs.
+
+        @param controller The controller that started this agent service.
+        @type controller: PlatformController
+
+        @return: the exit status code
+        @rtype: int
+        """
+
+        self.__start_time = time.time()
+        controller.register_for_termination(self.__handle_terminate)
+
+        # Register handler for when we get an interrupt signal.  That indicates we should dump the status to
+        # a file because a user has run the 'detailed_status' command.
+        self.__controller.register_for_status_requests(self.__report_status_to_file)
+
+        # noinspection PyBroadException
+        try:
+
+            self.__run_state = RunState()
+            self.__log_file_path = os.path.join(
+                self.__config.agent_log_path, AGENT_LOG_FILENAME
+            )
+            scalyr_logging.set_log_destination(
+                use_disk=True,
+                no_fork=self.__no_fork,
+                stdout_severity=self.__config.stdout_severity,
+                max_bytes=self.__config.log_rotation_max_bytes,
+                backup_count=self.__config.log_rotation_backup_count,
+                logs_directory=self.__config.agent_log_path,
+                agent_log_file_path=AGENT_LOG_FILENAME,
+            )
+
+            self.__update_debug_log_level(
+                self.__config.debug_level, self.__config.debug_level_logger_names
+            )
+
+            # We record where the log files(including multiprocess worker sessions logs) currently are
+            # so that we can (in the worse case) start copying them from those position.
+            logs_initial_positions = self.__get_log_files_initial_positions()
+
+            start_up_msg = scalyr_util.get_agent_start_up_message()
+            log.info(start_up_msg)
+            log.log(scalyr_logging.DEBUG_LEVEL_1, start_up_msg)
+
+            # Log warn message if non UTF-8 locale is used - this would cause issues when trying
+            # to monitor files with unicode characters inside the file names or inside the
+            # content
+            _, encoding, _ = scalyr_util.get_language_code_coding_and_locale()
+            if encoding.lower() not in ["utf-8", "utf8"]:
+                if sys.platform.startswith("win"):
+                    log.warn(NON_UTF8_LOCALE_WARNING_WINDOWS_MESSAGE % (encoding))
+                else:
+                    log.warn(NON_UTF8_LOCALE_WARNING_LINUX_MESSAGE % (encoding))
+
+            self.__controller.emit_init_log(log, self.__config.debug_init)
+
+            self.__start_or_stop_unsafe_debugging()
+
+            raw_scalyr_server = self.__config.raw_scalyr_server
+            self.__print_force_https_message(
+                self.__config.scalyr_server, raw_scalyr_server
+            )
+
+            last_config_change_check_time = (
```

Comment:
This double assignment with parenthesis got me a bit confused at first, but I assume black forces parenthesis so it's fine :)

Reply:
Yeah, I guess it's better to do a reformatting only after the review :)